tests/model_selection/scaffold_split.py

```diff
@@ -206,3 +224,85 @@ def test_scaffold_train_valid_test_split_return_indices(all_molecules):
     assert all(isinstance(idx, int) for idx in train_idxs)
     assert all(isinstance(idx, int) for idx in valid_idxs)
     assert all(isinstance(idx, int) for idx in test_idxs)
+
```

Comment:
Also add tests for raising errors - we have quite a few more situations where errors can be raised, and they should be covered by tests. You can use test coverage (see Makefile) to check which lines are not tested.